include/crocoddyl_msgs/inertial_parameters_publisher.h

```diff
@@ -0,0 +1,96 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2023-2024, Heriot-Watt University
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_PUBLISHER_H_
+#define CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_PUBLISHER_H_
+
+#include <realtime_tools/realtime_publisher.h>
+
+#ifdef ROS2
+#include <rclcpp/rclcpp.hpp>
+#else
+#include <ros/node_handle.h>
+#endif
+#include "crocoddyl_msgs/MultibodyInertialParameters.h"
+
+typedef crocoddyl_msgs::MultibodyInertialParameters MultibodyInertialParameters;
+
+namespace crocoddyl_msgs {
+
+class MultibodyInertialParametersRosPublisher {
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  typedef Eigen::Matrix<double, 10, 1> Vector10d;
+  /**
+   * @brief Initialize the multi-body inertial parameters publisher.
+   *
+   * @param[in] topic     Topic name
+   */
+  MultibodyInertialParametersRosPublisher(
+      const std::string &topic = "/crocoddyl/inertial_parameters")
+#ifdef ROS2
+      : node_("inertial_parameters_publisher"),
+        pub_(node_.create_publisher<MultibodyInertialParameters>(topic, 1)) {
+    RCLCPP_INFO_STREAM(node_.get_logger(),
+                       "Publishing MultibodyInertialParameters messages on "
+                           << topic);
+        }
+#else
+      {
+    ros::NodeHandle n;
+    pub_.init(n, topic, 1);
+    ROS_INFO_STREAM("Publishing MultibodyInertialParameters messages on "
+                    << topic);
+      }
+#endif
+      
+
+  ~MultibodyInertialParametersRosPublisher() = default;
+
+  /**
+   * @brief Publish a multi-body inertial parameters ROS message.
+   *
+   * @param parameters[in]    multibody inertial parameters
+   */
+  void
+  publish(const std::map<std::string, const Eigen::Ref<const Eigen::VectorXd> > &parameters
```

Comment:
We know it's always a 10-D vector, therefore, let's do
```suggestion
  publish(const std::map<std::string, const Eigen::Ref<const Vector10d> > &parameters
```